### DISCUSSION TITLE
Improved flexibility in the skyblock/auctions endpoint

### DISCUSSION THREAD

**Discussion by ghost** [HypixelDev/PublicAPI#373] [Ideas]
Currently the skyblock/auctions endpoint is extremely inflexible with no way to sort or request specific data. I think a good middle ground would be to provide more endpoints with cached data that divides the ah's info into categories such as bin/auctions, and maybe some sorted cashed pages such as by most recently listed, price, etc.  This would both cut down on usage of the api endpoints since if someone only wanted to get data on bins hypixel would only have to send half as many pages (or whatever the bin to auction ratio is) it would also be better for users since they wouldn't have to parse through data they didn't want anyway.  Also as a sidenote how is the current auctions data organized when using the skyblock/auctions endpoint?

**Comment by ghost** [comment]
Also as an extension, there is the auctions_ended to inform of auctions that ended in the past minute but is there anything for auctions started in the past minute?

**Reply by AzureAaron** [reply]
> Also as an extension, there is the auctions_ended to inform of auctions that ended in the past minute but is there anything for auctions started in the past minute?

There isn’t anything that I know of that’s public right now to retrieve auctions started in last minute.

They should really add auctions_started and auctions_modified to cut down on system resources and bandwidth though.

**Comment by ChiefChippy2** [comment]
IMO a 2-way websocket that sends event ( ie auction created, auction bidded, auction ended) to client and client can request data (ie auctions that started at x seconds ago) by sending a request to the server would be nice.
